subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/SnapshotHierarchy.java

```diff
@@ -41,6 +41,8 @@
             .map(CompleteFileSystemLocationSnapshot.class::cast);
     }
 
+    boolean hasDescendantSnapshot(String absolutePath);
```

Comment:
```suggestion
    boolean hasDescendantsUnder(String absolutePath);
```

WDYT? This is a _snapshot_ hierarchy, I think it would be good to not add "snapshot" in every method's name.

Reply:
Renamed to `hasDescendantsUnder`.